.gitignore

```diff
@@ -1,4 +1,31 @@
-.vs
-*.xproj.user
+[Oo]bj/
+[Bb]in/
+TestResults/
+.nuget/
+_ReSharper.*/
+packages/
+artifacts/
+PublishProfiles/
+*.user
+*.suo
+*.cache
+*.docstates
+_ReSharper.*
+nuget.exe
+*net45.csproj
+*net451.csproj
+*k10.csproj
+*.psess
+*.vsp
+*.pidb
+*.userprefs
+*DS_Store
+*.ncrunchsolution
+*.*sdf
+*.ipch
+*.sln.ide
+node_modules
+*.sln.ide
 project.lock.json
-npm-debug.log
+.vs/
+npm-debug.log
```

Comment:
About this file - it would be nice to keep it minimal. Could we limit these entries just to the files that appear without the developer's involvement on the two most common dev setups: VS and VS Code?